### PR TITLE
Add support to work with any RTMP server/ service

### PR DESCRIPTION
**Pull request by duggaraju**
Optimize to avoid encoding where possible and add support for more platforms/browsers.
Now works with iOS/Safari/Android/Chrome/Edge/Firefox.
Use copy codec of ffmpeg to transmux instead of transcode if the browser supports h264 and/or AAC natively.
Handle the case where browser produced resolution doesn't work fine with H264 since it is not an even number.
Include the current time in the video for easy latency check.
Works with YouTube/Facebook/Azure media services apart from Mux.

### CODE REVIEW

**Comment by duggaraju** [comment]
Add support for HTTP/HTTPS based on environment variable. Fix the dockerfile to work with legacy SSL provider since nodejs current version on alpine fails with "digital envelope routines::unsupported". Alternatively need to change Dockerfile to use nodejs:16-alpine (LTS)

**Comment by dylanjha** [comment]
Wow @duggaraju -- thank you for the contribution! On first glance, it all looks like good changes. Is there some documentation updates in the README that might be good to add? Particularly for the new query param options or env vars.

**Comment by duggaraju** [comment]
Sure. I can add documentation to the README.md to add steps to run with https (since http won't be able to access the camera when deployed to an external server).  However, regarding the query parameters to the WebSocket server I wasn't sure if that needs to be documented or if there is a good place for the same.

**Comment by duggaraju** [comment]
I updated the README and added a mobile screenshot as well (though can remove it if you don't want to).

**Comment by dylanjha** [comment]
Looks good @duggaraju -- is this still deployable to fly.io as-is? If you follow [these deploy instructions](https://github.com/MuxLabs/wocket#deploying-to-flyio) can it still be deployed with `flyctl deploy` ?

**Comment by duggaraju** [comment]
I haven't tried fly.io since I do not have an account but I tried using azure.   deploying is as simple as running
az containerapp up --name wocket --source .
I had to make small change in Docker to epxose the port and server.js to bind to 0.0.0.0 by default (instead of 127.0.0.1)

**Comment by dylanjha** [comment]
Okay -- I'll give fly a shot just because that's the default deployment instructions and we do talk about it in [this post on the fly blog](https://fly.io/blog/mux-fly-wocket-and-rtmp/)

As long as that still works then I'm happy to merge this! 🙌🏼

**Comment by dylanjha** [comment]
sorry, didn't meant to close -- testing this out with fly

**Comment by dylanjha** [comment]
@duggaraju I deployed to fly -- seems like everything is running but the live streaming is failing for me :( 

closing-or-closed_2022-07-14_12-37-33

Same happen for you (this is your branch)? https://purple-snow-7166.fly.dev/

Compare with current `master`: https://holy-frost-453.fly.dev/

I'll keep these apps up for a couple days before I delete them, but what I'm seeing is the streaming is not working on your branch, but it is working on current `master`

**Comment by duggaraju** [comment]
Interesting.  I just tested my branch with youtube and it works fine.  Seems to be an issue with the mux service. I am trying to optimize the backend server to only transcode when needed. So it only transcodes audio but uses video as it. Looks like mux service doesn't like that.  Maybe we can have a command line option to server to skip transcoding (default is to always transcode which is old behavior).  For example if you try the same from android phone or iphone you might get different behavior.

**Comment by duggaraju** [comment]
With the fix to always transcode by default I was able to get it working with mux as well.  Created a new environment variable SMART_TRANSCODE which if set to true  causes the new behavior with default being no transcode.

File: pages/index.js [hunk 1]
```diff
@@ -7,11 +7,32 @@ const CAMERA_CONSTRAINTS = {
   video: true,
 };
 
+const getRecoderSettings = () => {
```
Comment (johndeu) [inline pages/index.js]: Should be getRecorderSettings

**Comment by dylanjha** [comment]
@duggaraju confirmed that this works with Mux again 🙌🏼

**Comment by dylanjha** [comment]
Ready to merge this after fixing the typo @johndeu pointed out

**Comment by duggaraju** [comment]
Fixed the typo in the function name.

**Comment by dylanjha** [comment]
merged, thank you @duggaraju !

**Pull request merged**